__init__.py

```diff
@@ -42,51 +41,212 @@ class SetupState(str, Enum):
+            }
+        }
+        self.update_user_config(config)
+        self.create_dummy_identity()
+
+    # backend actions
+    @staticmethod
+    def create_dummy_identity():
+        # create pairing file with dummy data
+        login = {"uuid": uuid4(),
+                 "access": "OVOSdbF1wJ4jA5lN6x6qmVk_QvJPqBQZTUJQm7fYzkDyY_Y=",
+                 "refresh": "OVOS66c5SpAiSpXbpHlq9HNGl1vsw_srX49t5tCv88JkhuE=",
+                 "expires_at": time.time() + 999999}
+        IdentityManager.save(login)
+
+    @staticmethod
+    def update_device_attributes_on_backend():
+        """Communicate version information to the backend.
+
+        The backend tracks core version, enclosure version, platform build
+        and platform name for each device, if it is known.
+        """
+        LOG.info('Sending updated device attributes to the backend...')
+        try:
+            api = DeviceApi()
+            api.update_version()
+        except Exception:
+            pass
+
+
 class PairingSkill(OVOSSkill):
-    poll_frequency = 5  # secs between checking server for activation
 
     def __init__(self):
         super(PairingSkill, self).__init__("PairingSkill")
         self.reload_skill = False
-        self.api = DeviceApi()
-        self.data = None
-        self.time_code_expires = None
-        self.uuid = str(uuid4())
-        self.activator = None
-        self.activator_lock = Lock()
-        self.activator_cancelled = False
-        self.config_lock = Lock()
-        self.counter_lock = Lock()
-        self.count = -1  # for repeating pairing code. -1 = not running
-
         self.nato_dict = None
+        self.setup = None
+        self.pairing = None
         self.mycroft_ready = False
-        self.num_failed_codes = 0
-
-        self.state = SetupState.LOADING
+        self._state = SetupState.LOADING
+        self.pairing_mode = PairingMode.VOICE
 
     # startup
     def initialize(self):
-        # specific distros/vendors can override this
-        if "pairing_url" not in self.settings:
-            self.settings["pairing_url"] = "home.mycroft.ai"
-        if "color" not in self.settings:
-            self.settings["color"] = "#FF0000"
+        self.setup = SetupManager(self.bus)
+        self.pairing = PairingManager(self.bus, self.enclosure,
+                                      code_callback=self.on_pairing_code,
+                                      success_callback=self.on_pairing_success,
+                                      end_callback=self.on_pairing_end,
+                                      start_callback=self.on_pairing_start,
+                                      restart_callback=self.handle_pairing,
+                                      error_callback=self.on_pairing_error)
 
         self.add_event("mycroft.not.paired", self.not_paired)
-        self.add_event("ovos.setup.state", self.handle_get_setup_state)
+        self.add_event("ovos.setup.state.get", self.handle_get_setup_state)
```

Comment:
@AIIX not sure if you are using these messages, but pinging you because this changed from `ovos.setup.state` + `ovos.setup.state.response` to `ovos.setup.state.get` and `ovos.setup.state`